prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -68,7 +72,14 @@ class FHIRTranslator(
                 val receiver = settings.findReceiver(message.receiverFullName)
                     ?: throw RuntimeException("Receiver with name ${message.receiverFullName} was not found")
                 actionHistory.trackActionReceiverInfo(receiver.organizationName, receiver.name)
-                val bodyBytes = getByteArrayFromBundle(receiver, bundle)
+
+                val originalMessage = getOriginalMessage(message.reportId, WorkflowEngine())
```

Comment:
Can we move this inside the if below and change the if to:

`if (message.topic.isSendOriginal) {`

This way, we can save some computation if isSendOriginal is not true